examples/lines_bars_and_markers/stairs_demo.py

```diff
@@ -43,6 +43,15 @@
     ax.legend()
 plt.show()
 
+#############################################################################
+# *baseline* can take an array to allow for stacked histogram plots
+
+A = np.tile([1, 2, 3], (3, 1)) * np.arange(1, 4).reshape(-1, 1)
+A = np.vstack([np.zeros(3), A])
+
+for i in range(3):
+    plt.stairs(A[i+1], baseline=A[i], fill=True)
```

Comment:
```suggestion
for top, bottom in zip(A[1:], A[:-1]):
    plt.stairs(top, baseline=bottom, fill=True)
```

Reply:
It's more explicit, but is it really clearer?

Reply:
IMO it's more pythonic. I rarely use indices anymore. But that depends on where you're coming from. I'm also ok, if you want to leave the explicit index. But then please use `range(len(A) - 1)` to not have a magic number.

Reply:
I find the original much clearer...